docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes-provider.asciidoc

```diff
@@ -331,6 +331,81 @@ The policy generated by this configuration looks like:
         target: orchestrator
 ----
 
+[discrete]
+== Autodiscover targeted Pods using hints (standalone mode)
+
+Standalone Elastic Agent supports autodiscover based on hints from the provider.
+The hints system looks for hints in Kubernetes Pod annotations which have the prefix `co.elastic.hints`.
+As soon as the container starts, Elastic Agent will check if it contains any hints and launch the proper config for it.
+Hints tell Elastic Agent how to monitor the given container by using the proper integration.
+This is the full list of supported hints:
+
+[float]
+===== `co.elastic.hints/package`
+
+The package to use for monitoring.
+
+[float]
+===== `co.elastic.hints/data_stream`
+
+(optional) The list of data_streams to enable. If not specified the default data_streams will be enabled using the
+default values as specified in the respective package/integration.
+
+If data_streams are specified, then hints can be defined per data_stream like `co.elastic.hints/key.period: 10m`,
+otherwise the data_stream will use the top level hints (`co.elastic.hints/period: 10m`) in its configuration.
+
+[float]
+===== `co.elastic.hints/host`
+
+The host to use for metrics retrieval.
+
+[float]
+===== `co.elastic.hints/metrics_path`
+
+The path to retrieve the metrics from (/metrics by default).
+
+[float]
+===== `co.elastic.hints/period`
+
+The time interval for metrics retrieval, ie: 10s
+
+[float]
+===== `co.elastic.hints/timeout`
+
+Metrics retrieval timeout, default: 3s
+
+[float]
+===== `co.elastic.hints/username`
+
+The username to use for authentication
```

Comment:
username and password are required?
is it elasticsearch credentioals?

Reply:
I think these are integrations variables. If it is Redis for example , it should be Redis creds

Reply:
Yes, those are integration level settings. Keep in mind that most of them are taken directly from Beat's docs :) .

Reply:
it looks more like a required credentials, not as an configuration of specific integration 😕 
but, yes, I found the beats doc - https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-autodiscover-hints.html#_co_elastic_metricsusername